modules/oneIdIdSystem.js

```diff
@@ -0,0 +1,172 @@
+            reject(error);
+          }
+        });
+      }),
+    getId: () =>
+      new Promise((resolve, reject) => {
+        logInfo(`Taking oneId from 'sdk.getId'`);
+        sdk.firstParty.getId(resolve, reject);
+      }),
+  };
+};
+
+let loadPromise;
+const loadSDK = (w) => {
+  if (!loadPromise) {
+    loadPromise = new Promise((resolve, reject) => {
+      // In case it already loaded
+      if (w.OneIdSDK && w.OneIdSDK.firstParty) {
+        resolve(w.OneIdSDK);
+        return;
+      }
+      // In case it's not loaded yet
+      const oneIDSDKScript = document.createElement("script");
+      oneIDSDKScript.src = "https://cdn.oneid.live/sdk/oneIDSDK.js";
+      oneIDSDKScript.onabort = () => reject(Error("oneIDSDK loading aborted"));
+      oneIDSDKScript.oncancel = () =>
+        reject(Error("oneIDSDK loading canceled"));
+      oneIDSDKScript.onload = () => {
+        try {
+          w.OneIdSDK.firstParty.init(
+            {
+              fpVersion: "v4",
+              firstPartyURL: "",
+            },
+            () => {
+              if (w.OneIdSDK) resolve(w.OneIdSDK);
+              else reject(Error("OneIdSDK not loaded"));
+            }
+          );
+        } catch (error) {
+          reject(new Error("onload error: " + error.message));
+        }
+      };
+      document.body.appendChild(oneIDSDKScript);
+    });
+  }
+  return loadPromise;
+};
+
+/**
+ * @param {Object} [config]
+ */
+async function getOneId(config) {
+  const email = config.params.value;
+  const storageName = config.storage?.name;
+  let oneId =
+    storage.localStorageIsEnabled() && storageName
+      ? storage.getDataFromLocalStorage(storageName)
+      : null;
+
+  if (oneId) return oneId;
+  const w =
+    typeof window === "undefined"
+      ? globalThis
+      : window; /* && (window.top ? window.top : window); */
+  try {
+    const sdk = await loadSDK(w).then(createOneIdApi);
+    if (email) {
+      logInfo("Generating oneId from email");
+      const hem = await sdk.createHem(email);
+      oneId = await sdk.attachHem(hem);
+      logInfo("Passing oneId from email:", oneId);
+    } else {
+      logInfo("Generating oneId without email");
+      oneId = await sdk.getId();
+      logInfo("Passing oneId if from api:", oneId);
+    }
+    return oneId;
+  } catch (error) {
+    logError("Error generating oneId:", error);
+    throw error;
+  }
+}
+
+/** @type {Submodule} */
+export const oneIdIdSystem = {
+  /**
+   * used to link submodule with config
+   * @type {string}
+   */
+  name: MODULE_NAME,
+  /**
+   * decode the stored id value for passing to bid requests
+   * @function
+   * @param {string} value
+   * @returns {{oneId:string}}
+   */
+  decode(value) {
+    return { adrcid: value };
+  },
+  /**
+   * performs action to obtain id and return a value in the callback's response argument
+   * @function
+   * @param {SubmoduleConfig} [config]
+   * @returns {IdResponse|undefined}
+   */
+  getId(config) {
```

Comment:
do you need to provide any validations against user consentData?

Reply:
That makes sense.  If you guys already have something in place, sounds like you are already good there then.

Looks like there is a test failing within the circleci e2e test.. otherwise, everything else LGTM!